sdk/core/azure_core/src/http/poller.rs

```diff
@@ -437,50 +581,92 @@ enum State<N> {
+    assert!(
+        frequency >= MIN_RETRY_TIME,
+        "minimum polling frequency is 1 second"
+    );
+
+    let stream = unfold(
         // We flow the `make_request` callback through the state value to avoid cloning.
-        (State::Init, make_request),
-        |(state, make_request)| async move {
+        (State::Init, make_request, Some(target_tx)),
+        move |(state, make_request, target_tx)| async move {
             let result = match state {
                 State::Init => make_request(PollerState::Initial).await,
                 State::InProgress(n) => make_request(PollerState::More(n)).await,
                 State::Done => return None,
             };
             let (item, next_state) = match result {
-                Err(e) => return Some((Err(e), (State::Done, make_request))),
+                Err(e) => return Some((Err(e), (State::Done, make_request, target_tx))),
                 Ok(PollerResult::InProgress {
                     response,
                     retry_after,
                     next: n,
                 }) => {
                     // Note that test-proxy automatically adds a transform that zeroes an existing `after-retry` header during playback, so don't check at runtime:
                     // <https://github.com/Azure/azure-sdk-tools/blob/a80b559d7682891f36a491b73f52fcb679d40923/tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs#L1175>
-                    let duration = retry_after.unwrap_or(DEFAULT_RETRY_TIME);
+                    let duration = retry_after.unwrap_or(frequency);
 
                     tracing::trace!("retry poller in {}s", duration.whole_seconds());
                     sleep(duration).await;
 
                     (Ok(response), State::InProgress(n))
                 }
                 Ok(PollerResult::Done { response }) => (Ok(response), State::Done),
+                Ok(PollerResult::Succeeded {
+                    response,
+                    target: get_target,
+                }) => {
+                    // Send the target callback through the channel
+                    if let Some(tx) = target_tx {
+                        let _ = tx.send(get_target());
+                    }
+                    // Also yield the final status response and set target_tx to None since we consumed it
+                    return Some((Ok(response), (State::Done, make_request, None)));
+                }
             };
 
-            // Flow 'make_request' through to avoid cloning
-            Some((item, (next_state, make_request)))
+            // Flow 'make_request' and target_tx through to avoid cloning
+            Some((item, (next_state, make_request, target_tx)))
         },
-    )
+    );
+
+    let target = Box::new(async move {
+        match target_rx.await {
+            Ok(fut) => fut.await,
+            Err(_) => Err(crate::Error::new(
+                ErrorKind::Other,
+                "poller completed without sending target response",
```

Comment:
Shouldn't this indicate some kind of failure rather than just "completed without sending a response"?

Reply:
It's an apt description, and it's in an error. Much like Go, prefacing with "failed" is often unnecessary. Maybe I can reword a little.